src/OrchardCore.Modules/OrchardCore.Workflows/Trimming/Services/WorkflowTrimmingManager.cs

```diff
@@ -49,6 +49,7 @@ public async Task<int> PruneWorkflowInstancesAsync(TimeSpan retentionPeriod)
         var statuses = settings.Statuses.Select(x => (int)x).ToArray();
         var workflowInstances = await _session
             .Query<Workflow, WorkflowIndex>(x => x.WorkflowStatus.IsIn(statuses) && x.CreatedUtc <= dateThreshold)
+            .Take(batchSize)
```

Comment:
You may need to explicitly order the dataset to ensure you get all the records correctly on all DB engines